pkg/sync/sync_context.go

```diff
@@ -153,7 +153,10 @@ func WithResourceModificationChecker(enabled bool, diffResults *diff.DiffResultL
 	}
 }
 
-// WithNamespaceCreation will create a namespace, if it does not already exist
+// WithNamespaceCreation will create a namespace with the metadata passed in the `*unstructured.Unstructured` argument
+// of the `namespaceModifier` function, in the case it returns `true`. If the namespace already exists, the metadata
+// will overwrite what is already present if `namespaceModifier` returns `true`. If `namespaceModifier` returns `false`,
+// this will be a no-op.
 func WithNamespaceCreation(createNamespace bool, namespaceModifier func(*unstructured.Unstructured) bool) SyncOpt {
```

Comment:
In this case it is better to rename this function to `WithNamespaceModifier`. Just need to be careful to change all references. Also the `createNamespace` parameter seems it isn't making a lot of sense anymore.

Reply:
Done!